### PR TITLE
refactor!: upgrade to edgex v3

### PR DESCRIPTION
**Pull request by jrtitus**
### ⚠️ BREAKING CHANGE ⚠️ 

- upgrade all edgex modules to v3
- migrate configuration files from toml to yaml
- update configuration.yaml for use with common config
- update code and unit tests to support new sdk interface
- update docker base images and makefile instructions

### CODE REVIEW

File: cmd/res/configuration.yaml
Comment (nyameen) [inline cmd/res/configuration.yaml:1]: The config file is much lighter now.

File: internal/driver/driver.go [hunk 1]
```diff
@@ -14,65 +14,58 @@ import (
...
 
 // Initialize performs protocol-specific initialization for the device service
-func (d *Driver) Initialize(lc logger.LoggingClient, asyncCh chan<- *sdkModel.AsyncValues, deviceCh chan<- []sdkModel.DiscoveredDevice) error {
-	d.Logger = lc
-	d.AsyncCh = asyncCh
+func (d *Driver) Initialize(sdk interfaces.DeviceServiceSDK) error {
+	d.sdk = sdk
+
+	// Define custom API endpoints
+	if err := d.sdk.AddRoute("/api/v3/call", handleMethodCall, http.MethodPost); err != nil {
```
Comment (nyameen) [inline internal/driver/driver.go:45]: What is this `/call` for?

File: cmd/res/configuration.yaml
Comment (jrtitus) [reply cmd/res/configuration.yaml:1]: Yes, EdgeX took a lot of the configuration that was common to all services and set defaults in the common-config-bootstrapper service, so all service configurations will only include what is required to be set for them.

File: internal/driver/driver.go (hunk 1 above)
Comment (jrtitus) [reply internal/driver/driver.go:45]: Calling OPC UA methods associated with a given device. It was in the previous version - I just updated `/v2/` to `/v3/`. See [Using Methods](https://github.com/SE-I-T-Digital/device-opcua-go/tree/minnesota#using-methods) in the README

**Review by nyameen — APPROVED** [review]
LGTM

**Pull request merged**